plugins/jobtype/src/azkaban/jobtype/HadoopSparkJob.java

```diff
@@ -348,9 +414,51 @@ private static void handleStandardArgument(Props jobProps,
       }
     }
 
+    info("Final classpath: " + classPath);
     return classPath;
   }
 
+  private String[] getSparkHomeConf() {
+    String sparkHome = null;
+    String sparkConf = null;
+    // If user has specified version in job property. e.g. spark.version=1.6.0
+    String jobSparkVer = getJobProps().get(SparkJobArg.SPARK_VERSION.azPropName);
```

Comment:
Are you aware of the BangBang project? Let's discuss with @cwsteinbach about future directions on how to manage user specified dependencies.